documentation/prometheus-mixin/alerts.libsonnet

```diff
@@ -0,0 +1,176 @@
+{
+  prometheusAlerts+:: {
+    groups+: [
+      {
+        name: 'prometheus',
+        rules: [
+          {
+            alert: 'PrometheusBadConfig',
+            expr: |||
+              prometheus_config_last_reload_successful{%(prometheusSelector)s} == 0
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              message: 'Prometheus failed to reload config, see container logs',
+            },
+          },
+          {
+            alert: 'PrometheusNotificationQueueRunningFull',
+            expr: |||
+              predict_linear(prometheus_notifications_queue_length{%(prometheusSelector)s}[5m], 60 * 30)
+              >
+              prometheus_notifications_queue_capacity{%(prometheusSelector)s}
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              message: "Prometheus' alert notification queue is running full for {{$labels.namespace}}/{{ $labels.pod}}",
+            },
+          },
+          {
+            alert: 'PrometheusErrorSendingAlerts',
+            expr: |||
+              100 * rate(prometheus_notifications_errors_total{%(prometheusSelector)s}[5m])
+              /
+              rate(prometheus_notifications_sent_total{%(prometheusSelector)s}[5m]) > 1
```

Comment:
This makes it seem as though the > 1 is only filtering the rate.

```
    100 * rate(prometheus_notifications_errors_total{%(prometheusSelector)s}[5m])
  /
    rate(prometheus_notifications_sent_total{%(prometheusSelector)s}[5m])
> 
  1
```

Is the standard indentation, which makes this clear.

Reply:
I chose a slightly different indentation scheme (the outdented operator on its own line for vector-vector operations, operator with scalar on the same line for vector-scalar operations).

Is the "standard indentation" documented anywhere?

Reply:
It's documented by example at https://prometheus.io/docs/practices/rules/

Reply:
Interesting. It also uses the parentheses hack, although Yaml has a better way of handing the indent in the 1st line.

In any case, the examples do not contain vector-to-scalar ops.